source/agora/common/TransactionData.d

```diff
@@ -0,0 +1,284 @@
+/*******************************************************************************
+
+    Contains a storage class consisting of a maximum size fixed byte array.
+
+    Copyright:
+        Copyright (c) 2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.common.TransactionData;
+
+import agora.common.Hash;
+import agora.common.Serializer;
+
+static import std.ascii;
+import std.conv: parse;
+import std.algorithm.iteration : each, map;
+import std.range;
+import std.utf;
+
+
+public struct TransactionData
+{
+    invariant()
+    {
+        assert(length <= MAX_SIZE);
+    }
+
+    /// The maximum size
+    public static immutable MAX_SIZE = 1024;
+
+    /// The byte array of transaction data
+    public ubyte[MAX_SIZE] data;
+
+    /// The number of bytes of transaction data
+    public size_t length;
+
+    /// If the input data is larger than MAX_SIZE, this value is true, if not false.
+    public bool is_oversize;
```

Comment:
What's the reasoning for such a flag ?

Reply:
If the data is larger than the maximum, this transaction will be ignored.
I did this to eliminate factors that could be a problem with the stability of the system due to the excessive waste of memory.